src/onnx/parse_layernorm.cpp

```diff
@@ -112,16 +113,23 @@ struct parse_layernorm : op_parser<parse_layernorm>
         if(skipped_axes > 0)
         {
             auto x_dims = x_shape.lens();
-            scale_bcast = info.add_instruction(
-                make_op("broadcast", {{"axis", skipped_axes}, {"out_lens", x_dims}}), scale);
+            if(scale->get_shape().ndim() != x_rank)
+            {
+                scale_bcast = info.add_instruction(
+                    make_op("broadcast", {{"axis", skipped_axes}, {"out_lens", x_dims}}), scale);
```

Comment:
I think the broadcast is only needed when its a 1d tensor.

Reply:
updated